pkg/controller/expcertificates/readiness/readiness_controller.go

```diff
@@ -148,12 +148,33 @@ func (c *controller) ProcessItem(ctx context.Context, key string) error {
 	crt = crt.DeepCopy()
 	apiutil.SetCertificateCondition(crt, condition.Type, condition.Status, condition.Reason, condition.Message)
 
-	if input.Secret != nil && input.Secret.Data != nil {
+	// set Certificate status fields based on the contents of the x509 certificate
+	// if the certificate is not issued, or the data cannot be decoded, the status
+	// field values are cleared.
+	clearCertificateFields := func() {
+		crt.Status.NotAfter = nil
+		crt.Status.NotBefore = nil
+		crt.Status.RenewalTime = nil
+	}
+	switch {
+	case input.Secret != nil && input.Secret.Data != nil:
 		x509cert, err := pki.DecodeX509CertificateBytes(input.Secret.Data[corev1.TLSCertKey])
-		if err == nil {
-			t := metav1.NewTime(x509cert.NotAfter)
-			crt.Status.NotAfter = &t
+		if err != nil {
+			clearCertificateFields()
+			break
```

Comment:
This can also be caused by users providing their own Secret resources with invalid data. We previously did not log an error here either as this is the 'readiness' controller, which is one of many that each independently operate on the Certificate resource, meaning we could print errors each time it was synced up until the point that it's issued.